### PR TITLE
feat: add indexing to endpoints

### PR DESCRIPTION
**Pull request by garrensmith**
## Describe your changes
Adds a new explain endpoint
Adds indexes to the describe collection endpoint

The explain endpoint is very basic. I want to add the key ranges and the actual index used. This will be done in a later
PR. I want to do some code refactoring before I add that part.

### CODE REVIEW

File: server/v1/api.proto [hunk 1]
```diff
@@ -381,6 +381,8 @@ message ReadRequest {
   bytes sort = 6;
   // Optionally specify a database branch name to perform operation on
   string branch = 7;
+	// Return an explanation of what Tigris would do to complete the read query
+	bool explain = 8;
```
Comment (himank) [inline server/v1/api.proto:385]: is formatting off?

File: server/v1/api.proto [hunk 2]
```diff
@@ -390,6 +392,15 @@ message ReadResponse {
   bytes resume_token = 2;
   // Has metadata related to the documents stored.
   ResponseMetadata metadata = 3;
+
+  ExplainResponse explain = 4;
+}
+
+message ExplainResponse {
+  string collection = 1;
```
Comment (himank) [inline server/v1/api.proto:400]: why do we need this?

File: server/v1/api.proto [hunk 3]
```diff
@@ -390,6 +392,15 @@ message ReadResponse {
...
+  string index = 2;
```
[the remark this answers is not in the document]
Comment (garrensmith) [reply server/v1/api.proto:401]: Maybe but we would split it by stages.

File: server/v1/api.proto (hunk 2 above)
Comment (garrensmith) [reply server/v1/api.proto:397]: It gives the name of the collection used. I've based that off the fact Mongo includes that in its explain response

File: server/v1/api.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (himank) [reply server/v1/api.proto]: I mean this is fine to be string, but if you need to return multiple indexes in the response then you will need to add one more field?

File: server/v1/api.proto (hunk 2 above)
Comment (himank) [reply server/v1/api.proto:397]: sounds good then let's keep it.

**Review by himank — APPROVED** [review]

**Pull request merged**